java/test/jmri/jmrix/can/cbus/swing/modeswitcher/SprogCbusModeSwitcherPaneTest.java

```diff
@@ -0,0 +1,63 @@
+package jmri.jmrix.can.cbus.swing.modeswitcher;
+
+import java.awt.GraphicsEnvironment;
+import jmri.GlobalProgrammerManager;
+import jmri.jmrix.can.CanSystemConnectionMemo;
+import jmri.jmrix.can.TrafficControllerScaffold;
+import jmri.jmrix.can.cbus.CbusDccProgrammer;
+import jmri.jmrix.can.cbus.CbusDccProgrammerManager;
+import jmri.jmrix.can.cbus.CbusPreferences;
+import jmri.util.JUnitUtil;
+import org.junit.After;
+import org.junit.Assume;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Tests for the ModeSwitcherPane class
+ *
+ * @author Andrew Crosland (C) 2020
+ */
+public class SprogCbusModeSwitcherPaneTest extends jmri.util.JmriJFrameTestBase {
+
+    CanSystemConnectionMemo memo;
+    CbusDccProgrammer prog;
+    jmri.jmrix.can.TrafficController tc;
+    
+    @Test
+    public void testInitComponents() throws Exception{
+        Assume.assumeFalse(GraphicsEnvironment.isHeadless());
+        // for now, just make sure there isn't an exception.
+        ((SprogCbusModeSwitcherPane) frame).initComponents();
+    }
+
+    @Before
+    @Override
+    public void setUp() {
+        JUnitUtil.setUp();
+        
+        jmri.InstanceManager.setDefault(jmri.jmrix.can.cbus.CbusPreferences.class,new CbusPreferences() );
+
+        tc = new TrafficControllerScaffold();
+        memo = new CanSystemConnectionMemo();
+        memo.setTrafficController(tc);
+        prog = new CbusDccProgrammer(tc);
+        
+        jmri.InstanceManager.setDefault(GlobalProgrammerManager.class,new CbusDccProgrammerManager(prog, memo) );
+        if (!GraphicsEnvironment.isHeadless()) {
+            frame = new SprogCbusModeSwitcherPane(memo);
+        }
+    }
+
+    @After
+    @Override
+    public void tearDown() {
+        prog = null;
+        tc.terminateThreads();
+        memo.dispose();
+        tc = null;
+        memo = null;
+        JUnitUtil.resetWindows(false,false);
+        super.tearDown();
```

Comment:
You don’t want to call JUnitUtil.resetWindows before the call to super.tearDown.  super.tearDown disposes of the frame and then calls resetWindows.

Reply:
OK, I added that as I saw it used in another test and thought it may help.